compiler/ballerina-lang/src/main/java/io/ballerina/projects/Project.java

```diff
@@ -78,6 +80,14 @@ public Path sourceRoot() {
         return this.sourceRoot;
     }
 
+    public Path targetDir() {
+        if (this.buildOptions == null || this.buildOptions.getTargetPath() == null) {
```

Comment:
We dont need a null check for buildOptions AFAIU. A project is alway created with buildOptions.